pandas/tests/arrays/categorical/test_constructors.py

```diff
@@ -535,24 +535,15 @@ def test_from_codes_with_nan_code(self):
         with pytest.raises(ValueError, match="codes need to be array-like integers"):
             Categorical.from_codes(codes, dtype=dtype)
 
-    def test_from_codes_with_float(self):
+    @pytest.mark.parametrize("codes", [[1.0, 2.0, 0], [1.1, 2.0, 0]])
+    def test_from_codes_with_float(self, codes):
         # GH21767
-        codes = [1.0, 2.0, 0]  # integer, but in float dtype
         dtype = CategoricalDtype(categories=["a", "b", "c"])
 
-        # empty codes should not raise for floats
```

Comment:
can you retain this comment and the one on L552 describing the choice of parametrizations

Reply:
Do you mean the inline comments from L540 and L552? This comment (L543) I find a little confusing because we're testing an empty list rather than an empty float array

Reply:
thanks